metering/reports/metering-writing-custom-report-queries.adoc

```diff
@@ -0,0 +1,396 @@
+--------+-------------------------+---------------+------------------------------------------------------------------+------------
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=telemeter-tschuy, deployment=presto-worker}           | 2019-05-03
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=metering-emoss, deployment=metering-operator}         | 2019-05-03
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=openshift-monitoring, deployment=prometheus-operator} | 2019-05-03
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=metering-tschuy, deployment=presto-worker}            | 2019-05-03
+    0.0 | 2019-05-03 16:34:00.000 |          60.0 | {namespace=telemeter-tschuy, deployment=presto-worker}           | 2019-05-03
+    1.0 | 2019-05-03 16:18:00.000 |          60.0 | {namespace=metering-tschuy, deployment=reporting-operator}       | 2019-05-03
+    0.0 | 2019-05-03 16:12:00.000 |          60.0 | {namespace=telemeter-tschuy, deployment=presto-worker}           | 2019-05-03
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=metering-tflannag, deployment=reporting-operator}     | 2019-05-03
+    0.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=metering-tflannag, deployment=presto-worker}          | 2019-05-03
+    1.0 | 2019-05-03 16:01:00.000 |          60.0 | {namespace=metering-chancez, deployment=reporting-operator}      | 2019-05-03
+(10 rows)
+----
+
+Now it is time to start talking about our data in terms of SQL.
+As you can see, our new table has 4 columns which are documented in the ReportDataSource Table Schema documentation.
+
+== Writing Presto queries against PrometheusMetricsImporter ReportDataSource Data
+
+Now that we have a database table that we can experiment with, we can begin to answer the question we started with:
+
+_What is the average and total amount of time that a particular deployment's replicas are unready?_
+
+To answer this we need to do a few things:
+
+* For each timestamp, find the time each individual pod was unready.
+* Divide up the results by the deployment.
+* Find the average and total duration pods are unready for each deployment.
+
+We start with getting the unready time at an individual metric level for each timestamp.
+Since the `amount` corresponds to the number of unready pods at that moment in time, and the `timeprecision` gives how long the metric was that value, we just need to multiply the `amount` (number of pods) by the `timeprecision` (length of time it was at that value):
+
+[source,sql]
+----
+SELECT
+    "timestamp",
+    labels['namespace'] as namespace,
+    labels['deployment'] as deployment,
+    amount * "timeprecision" as pod_unready_seconds
+FROM datasource_your_namespace_unready_deployment_replicas
+ORDER BY pod_unready_seconds DESC, namespace ASC, deployment ASC
+LIMIT 10;
+----
+
+Next, we need to group our results by the deployment, this can be done using a SQL `GROUP BY` clause.
```

Comment:
s/Next, we need to group our results by the deployment, this can be done using/group your results by deployment using